File: CleverOnion-backend/src/main/java/com/cleveronion/blog/common/exception/ResourceNotFoundException.java
package com.cleveronion.blog.common.exception;

/**
 * 资源未找到异常
 * 当请求的资源不存在时抛出此异常
 * 
 * @author CleverOnion
 * @since 1.0.0
 */
public class ResourceNotFoundException extends BusinessException {
    
    /**
     * 构造函数
     * 
     * @param message 错误消息
     */
    public ResourceNotFoundException(String message) {
        super("RESOURCE_NOT_FOUND", message);
    }
    
    /**
     * 构造函数
     * 
     * @param resourceType 资源类型
     * @param resourceId 资源ID
     */
    public ResourceNotFoundException(String resourceType, Object resourceId) {
        super("RESOURCE_NOT_FOUND", String.format("%s with id '%s' not found", resourceType, resourceId));
    }
    
    /**
     * 构造函数
     * 
     * @param message 错误消息
     * @param cause 原因
     */
    public ResourceNotFoundException(String message, Throwable cause) {
        super("RESOURCE_NOT_FOUND", message, cause);
    }
}

File: CleverOnion-blog-k/src/main/kotlin/com/cleveronion/application/shared/exception/ResourceNotFoundException.kt
package com.cleveronion.application.shared.exception

/**
 * 资源未找到异常
 * 
 * 当应用层无法找到请求的资源时抛出此异常
 */
class ResourceNotFoundException(
    message: String,
    val resourceType: String? = null,
    val resourceId: String? = null,
    cause: Throwable? = null
) : ApplicationException(message, cause) {
    
    companion object {
        /**
         * 创建资源未找到异常
         */
        fun of(resourceType: String, resourceId: String): ResourceNotFoundException {
            return ResourceNotFoundException(
                message = "$resourceType with id '$resourceId' not found",
                resourceType = resourceType,
                resourceId = resourceId
            )
        }
        
        /**
         * 创建通用资源未找到异常
         */
        fun of(message: String): ResourceNotFoundException {
            return ResourceNotFoundException(message)
        }
    }
}